KOGITO-7910: Patch PR for Kubernetes service discovery guide with doc improvements

hmanwani-rh opened the pull request:
<!-- Please don't forget your JIRA link -->
**JIRA:** https://issues.redhat.com/browse/KOGITO-7910

<!-- If you don't have a JIRA link, please provide a short description of what this PR does -->
**Description:** Patch PR for Kubernetes service discovery guide with doc improvements

<!-- Link to related PRs: -->
https://github.com/kiegroup/kogito-docs/pull/177

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category
- [x] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to setup automatic cherry-pick to another branch ?
</summary>

The cherry-pick action allows to setup automatic cherry-pick from `main` to a specific branch.

To allow it, you will need to add the corresponding label with pattern: `backport-{RELEASE_BRANCH}`.  
For example, if a backport to branch `1.26.x` is needed, then the label should be `backport-1.26.x`.

Once the PR is merged, the action will retrieve the commit and cherry-pick it to the desired branch.

*NOTE: You can still add label after the merge and it should still be cherry-picked.*
</details>

hmanwani-rh commented:
@spolti Please review the changes in this PR and share your comments.

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -1,18 +1,16 @@
-then you should consider to use a static URL instead. The following sketch demonstrates how this URI is composed:
+The Kubernetes service discovery feature works during the workflow application startup, in which this feature scans all the Quarkus configuration in search of the following URI pattern:
 
+.URI pattern in Kubernetes service discovery
```
```suggestion
.KubeURI Pattern
```

ricardozanini replied:
Please don't use `KubeURI`, this means nothing.

hmanwani-rh replied:
So should we keep `.URI pattern in Kubernetes service discovery`?

spolti commented on serverlessworkflow/modules/ROOT/nav.adoc:
```diff
@@ -57,7 +57,7 @@
 // ** xref:cloud/deploying-on-kubernetes-cluster.adoc[Deploying on Kubernetes Clusters]
 // ** xref:cloud/versioning-workflows-in-knative.adoc[Versioning workflows in Knative]
-** xref:cloud/kubernetes-service-discovery.adoc[Kubernetes Service Discovery for Serverless Workflow Application]
+** xref:cloud/kubernetes-service-discovery.adoc[Kubernetes service discovery in {context}]
```
not sure what context will be translated to, wouldn't be better just use Kubernetes service discovery?

hmanwani-rh replied:
the {context} will be translated to Serverless Workflow in upstream and OpenShift Serverless logic in downstream.

spolti replied:
okay, thanks.

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
 ----
 
+[[ref-example-kubernetes-service-discovery]]
+== Example of Kubernetes service discovery in {context}
```
which example are you referring to?

hmanwani-rh replied:
referring to the `serverless-workflow-greeting-quarkus` example, which is also used in the URI in the section.

spolti replied:
not sure if I understand it.

hmanwani-rh replied:
We have a URI in this section as follows which contains `serverless-workflow-greeting-quarkus`, therefore added example:

![image](https://user-images.githubusercontent.com/59050394/190379037-96714827-b0ce-4aac-a156-530d0bc17233.png)

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
 ----
 
-The following scheme values are supported:
+Therefore, you must keep in mind that if application startup time matters, then consider to use a static URL instead. 
```
isn't this sentence lost here?

hmanwani-rh replied:
Done, changed the paragraph above the URI pattern as follows:

The Kubernetes service discovery feature works during the workflow application startup, in which this feature scans all the Quarkus configuration in search of the URI pattern. Therefore, you must keep in mind that if application startup time matters, then consider to use a static URL instead.

Following is the URI pattern in Kubernetes service discovery:

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
 
-Also known as Query String, they are defined in the very same way that we are already used to do with URLs to assign value
-to specific attributes.
+Query parameters in URI::
```
```suggestion
KubeURI Query Parameters::
```

ricardozanini replied:
Same thing. This is just a URI that represents a Kubernetes object in the cluster.

hmanwani-rh replied:
So, should we keep as-is?

spolti commented:
Why the job that renders the page wasn't  triggered?

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -130,28 +131,20 @@ __  ____  __  _____   ___  __ ____  ______
-Note that the URI was translated to `http://greeting-quarkus-cli.serverless-workflow-greeting-quarkus.10.99.154.147.sslip.io` when the application started, and will be used at runtime any time it needs it.
+In the previous example, the URI is translated to `http://greeting-quarkus-cli.serverless-workflow-greeting-quarkus.10.99.154.147.sslip.io` when the application started. The translated URI is used at runtime, when needed.
 ====
-== Discovery Precedence
```
I would keep it.

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
-to specific attributes.
+Query parameters in URI::
++
+--
```
better to keep or add a small tittle here, appending information for different contexts may lead to some confusion.

hmanwani-rh replied:
Removing other information from this section.

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
-Then if at least one is found, the engine will parse the URI, query the Kubernetes API for the given resource, and override the given property.
-
-Suppose we have an application that will consume a resource running on Kubernetes. This resource is a Knative Service that exposes a function that can be discovered with the following URI:
+The Kubernetes service discovery is performed at the _STATIC_INIT_ time of Quarkus during the workflow application startup. First, the service discovery scans the Quarkus configuration values and searches for the Kubernetes URI pattern. If the URI pattern is found, the engine parses the URI, queries the Kubernetes API for the given resource, and overrides the given property.
```
```suggestion
The Kubernetes service discovery is performed at the _STATIC_INIT_ time of Quarkus during the workflow application startup. First, the service discovery scans the Quarkus configuration values and searches for the Kubernetes URI pattern. If the URI pattern is found, the engine parses the URI, queries the Kubernetes API searching for the given resource, and overrides the given property.
```

hmanwani-rh replied:
Done

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
-Then if at least one is found, the engine will parse the URI, query the Kubernetes API for the given resource, and override the given property.
-
-Suppose we have an application that will consume a resource running on Kubernetes. This resource is a Knative Service that exposes a function that can be discovered with the following URI:
+The Kubernetes service discovery is performed at the _STATIC_INIT_ time of Quarkus during the workflow application startup. First, the service discovery scans the Quarkus configuration values and searches for the Kubernetes URI pattern. If the URI pattern is found, the engine parses the URI, queries the Kubernetes API for the given resource, and overrides the given property.
```
```suggestion
The Kubernetes service discovery is performed at the _STATIC_INIT_ time of Quarkus during the workflow application startup. First, the service discovery scans the Quarkus configuration values and searches for the Kubernetes URI pattern. If the URI pattern is found, the engine parses the URI, queries the Kubernetes API for the given resource, and overrides the given application property.
```

hmanwani-rh replied:
Done

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
+image::cloud/sw-discovery-flow.jpg[]
 
-== Configuration
+There is no specific configuration required for the Kubernetes service discovery except by using the expected URI pattern. However, the `okhttp` communication interceptor, which logs the communication between the application and the Kubernetes API is disabled by default.
```
also, these following sentences are a little bit disconnected without a tittle, maybe keep the Configuration subtitle and group it and the dependency exclusion?

hmanwani-rh replied:
Added a section for configuration.

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
-There is no specific configuration for the Discovery to work properly except by using the expected URI pattern.
-However, the okhttp communication interceptor, which logs the communication between the application and the Kubernetes API is disabled by default. +
-If there is a need to debug it, it can be enabled just by setting this application property:
+You can enable the `okhttp` communication interceptor if there is a need to debug it by setting the following application property:
```
```suggestion
You can enable the `okhttp` communication interceptor if there is a need to debug communication betwwen the client and the Kubernetes API by setting the following application property:
```

hmanwani-rh replied:
Done

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
-If there is a need to debug it, it can be enabled just by setting this application property:
+You can enable the `okhttp` communication interceptor if there is a need to debug it by setting the following application property:
 
+.Property to enable `okhttp` communication interceptor
```
```suggestion
.Application Property to enable `okhttp` communication interceptor
```

hmanwani-rh replied:
Done

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -130,28 +131,20 @@ __  ____  __  _____   ___  __ ____  ______
 [tabs]
 ====
-Service Discovery Enabled::
+Kubernetes service discovery is enabled::
```
isn't better to remove the "is"?

hmanwani-rh replied:
I think the sentence looks better with is.

spolti commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -130,28 +131,20 @@ __  ____  __  _____   ___  __ ____  ______
 (main) sw-service-discovery 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 2.360s. Listening on: http://0.0.0.0:8080
 ----
-Service Discovery Disabled::
+Kubernetes service discovery is disabled::
```
same

hmanwani-rh replied:
I think the sentence looks better with is.

spolti reviewed (commented):
Thanks @hmanwani-rh 
I've added a few comments.

hmanwani-rh commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
+Query parameters in URI::
++
+--
+The query parameters are also known as query string. The query parameters are defined the similar way with URLs to assign value to specific attributes.
```
Do we need to capitalize Query String?

ricardozanini replied:
I don't think so. Indeed in the RFC is just `query`. https://www.rfc-editor.org/rfc/rfc2616#section-3.2.2

But I think `query string` is just fine.

spolti replied:
> Do we need to capitalize Query String?

no.

hmanwani-rh replied:
Changed it to "Also known as query string`.

hmanwani-rh commented:
> Why the job that renders the page wasn't triggered?

Working now

hmanwani-rh commented on serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc:
```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
-* **Custom port name**: Useful to determine which port to use when there are many ports configured in the target Service or Container.
-It can be configured using the following patter:
-** `port-name=<PORT_NAME>`: the port name to be queried.
+Based on the resource to be discovered, the Kubernetes service discovery follows specific paths as shown in the following figure:
```
Done, changed it to a point.

hmanwani-rh commented:
> Thanks @hmanwani-rh I've added a few comments.

@spolti Thanks for reviewing the changes and adding comments. I've incorporated the changes as suggested. Please take a look and let me know if the changes look good.

spolti reviewed (approved):
Many thanks @hmanwani-rh 
feel free to merge.
